examples/robot-assistant/server/src/main.rs

```diff
@@ -0,0 +1,128 @@
+                    println!("{}", input);
+                    producer.send_record(&input, 0).await?;
+                }
+                Ok(())
+            });
+            let topic_id_clone = topic_id.clone();
+            let consume_handle = spawn(async move {
+                let consumer = fluvio::consumer(topic_id_clone, 0).await?;
+                let mut fluvio_stream = consumer.stream(fluvio::Offset::beginning()).await?;
+                while let Some(Ok(record)) = fluvio_stream.next().await {
+                    if let Some(bytes) = record.try_into_bytes() {
+                        let s = String::from_utf8_lossy(&bytes);
+                        ws_stream
+                            .send_string(format!("<div style=\"text-align: right;\">{}</div>", s))
+                            .await
+                            .map_err(|_| Error::new(ErrorKind::Other, "oh no!"))?;
+                        let message = if let Ok(num) = s.parse::<usize>() {
+                            robot::Message::Number(num)
+                        } else {
+                            robot::Message::Text(s.to_string())
+                        };
+                        let state = robot.process(message);
+                        match state {
+                            State::Text { prompt, .. } => ws_stream
+                                .send_string(format!("<div class=\"bot-message\">{}</div>", prompt))
+                                .await
+                                .map_err(|_| Error::new(ErrorKind::Other, "oh no!"))?,
+                            State::Number { prompt, items } => {
+                                ws_stream
+                                    .send_string(format!(
+                                        "<div class=\"bot-message\">{}</div>",
+                                        prompt
+                                    ))
+                                    .await
+                                    .map_err(|_| Error::new(ErrorKind::Other, "oh no!"))?;
+                                for item in items {
+                                    ws_stream
+                                        .send_string(format!(
+                                            "<div>[ {} ] {}</div>",
+                                            item.next, item.answer
+                                        ))
+                                        .await
+                                        .map_err(|_| Error::new(ErrorKind::Other, "oh no!"))?;
+                                }
+                            }
+                            State::End { message } => ws_stream
+                                .send_string(format!("<div>{}</div>", message))
+                                .await
+                                .map_err(|_| Error::new(ErrorKind::Other, "oh no!"))?,
+                        }
+                    }
+                }
+                Ok(())
+            });
```

Comment:
Can you extract the body inside this `spawn` into its own function?

Reply:
I am not sure how to do this in rust, the async block also capture the ws_stream, which is outside the spawn block.